### PR TITLE
blacklist/priority makeover, refactor base_miner to /_neuron

### PR DESCRIPTION
**Pull request by ifrit98**
- separates out _blacklist and _priority into their own dirs.
- clean up blacklist, add vpermit check
- move base_prompting_miner.py to _neuron.

### CODE REVIEW

File: bittensor/__init__.py [hunk 1]
```diff
@@ -207,7 +209,9 @@ def turn_console_off():
 from bittensor._dendrite.text_prompting.dendrite_pool import TextPromptingDendritePool as text_prompting_pool
 
 # ---- Base Miners -----
-from bittensor._synapse.text_prompting.miner import BasePromptingMiner
+from bittensor._neuron.base_miner_neuron import BaseMinerNeuron as base_miner_neuron
```
Comment (Eugene-hu) [inline bittensor/__init__.py]: small nit: BasePromptingMiner and base_miner_neuron are two separate conventions. We should stick with one, ie BaseMinerNeuron.

File: bittensor/_neuron/base_validator.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+import torch
+import threading
+import argparse
+import bittensor
+
+from rich import print
+from typing import List, Dict, Union, Tuple, Optional
+from datetime import datetime
+
+class BaseValidator:
```
Comment (Eugene-hu) [inline bittensor/_neuron/base_validator.py:29]: I am little confused about the inheritance here; what is the use case for this BaseValidator? is it being inherited by the core_validator?

File: bittensor/_neuron/base_miner_neuron.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+import torch
+import threading
+import argparse
+import bittensor
+
+from rich import print
+from typing import Union, Tuple
+from datetime import datetime
+
+class BaseMinerNeuron:
```
Comment (Eugene-hu) [inline bittensor/_neuron/base_miner_neuron.py:29]: Similar question here; is this the base miner for the servers?

Comment (ifrit98) [reply bittensor/_neuron/base_miner_neuron.py:29]: We will have:
- TextToImage
- SpeechToText
- TextToCompletion (This is currently text_prompting and needs to extend this, working on it now)

So the main idea is to extend each multi-modal class from this `BaseMinerNeuron` that way we don't have entirely bespoke submodules.

File: bittensor/__init__.py (hunk 1 above)
Comment (ifrit98) [reply bittensor/__init__.py]: Fixed with commit: https://github.com/opentensor/bittensor/commit/07fad43a72bcdcd3c941ef84ca39c72db2e1c9a3

File: bittensor/_neuron/base_validator.py (hunk 2 above)
Comment (ifrit98) [reply bittensor/_neuron/base_validator.py:29]: Yes, it should be inherited by all future Validator subtypes. (see next comment for details)

**Review by Eugene-hu — APPROVED** [review]

**Review by Eugene-hu — APPROVED** [review]

**Pull request merged**